### PR TITLE
SEO and site search improvements

### LINKED ISSUE #34347: SEO: Sweep all of the .NET fundamentals
Focus on the following two areas of SEO optimization:

- Keyword Research: Identify the keywords and phrases that your target audience is searching for in search engines. Use tools like Google Keyword Planner, Ahrefs, or SEMrush to find the best keywords for your content.

- On-Page Optimization: Optimize your website's structure, meta tags, titles, and descriptions for the identified keywords. Also, optimize your content, including headings, subheadings, images, and internal links, to make it more readable and relevant to the audience.

Follow [internal guidance](https://review.learn.microsoft.com/en-us/help/contribute/contribute-seo-onboarding?branch=main) on driving updates.

---
[Associated WorkItem - 67210](https://dev.azure.com/msft-skilling/Content/_workitems/edit/67210)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

SEO and site search improvements

Fixes #34347

#### Internal previews

<details><summary><strong>Toggle Expand/Collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/extensions/best-practices-for-developing-world-ready-apps.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/best-practices-for-developing-world-ready-apps.md) | [docs/core/extensions/best-practices-for-developing-world-ready-apps](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/best-practices-for-developing-world-ready-apps?branch=pr-en-us-34538) |
| [docs/core/extensions/caching.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/caching.md) | [docs/core/extensions/caching](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/caching?branch=pr-en-us-34538) |
| [docs/core/extensions/channels.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/channels.md) | [docs/core/extensions/channels](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/channels?branch=pr-en-us-34538) |
| [docs/core/extensions/cloud-service.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/cloud-service.md) | [docs/core/extensions/cloud-service](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/cloud-service?branch=pr-en-us-34538) |
| [docs/core/extensions/configuration-providers.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/configuration-providers.md) | [docs/core/extensions/configuration-providers](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/configuration-providers?branch=pr-en-us-34538) |
| [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/configuration.md) | [docs/core/extensions/configuration](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/configuration?branch=pr-en-us-34538) |
| [docs/core/extensions/console-log-formatter.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/console-log-formatter.md) | [docs/core/extensions/console-log-formatter](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/console-log-formatter?branch=pr-en-us-34538) |
| [docs/core/extensions/create-resource-files.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/create-resource-files.md) | [docs/core/extensions/create-resource-files](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/create-resource-files?branch=pr-en-us-34538) |
| [docs/core/extensions/create-satellite-assemblies.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/create-satellite-assemblies.md) | [docs/core/extensions/create-satellite-assemblies](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/create-satellite-assemblies?branch=pr-en-us-34538) |
| [docs/core/extensions/custom-configuration-provider.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/custom-configuration-provider.md) | [docs/core/extensions/custom-configuration-provider](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/custom-configuration-provider?branch=pr-en-us-34538) |
| [docs/core/extensions/custom-logging-provider.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/custom-logging-provider.md) | [docs/core/extensions/custom-logging-provider](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider?branch=pr-en-us-34538) |
| [docs/core/extensions/dependency-injection-guidelines.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/dependency-injection-guidelines.md) | [docs/core/extensions/dependency-injection-guidelines](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines?branch=pr-en-us-34538) |
| [docs/core/extensions/dependency-injection-usage.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/dependency-injection-usage.md) | [docs/core/extensions/dependency-injection-usage](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage?branch=pr-en-us-34538) |
| [docs/core/extensions/dependency-injection.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/dependency-injection.md) | [docs/core/extensions/dependency-injection](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection?branch=pr-en-us-34538) |
| [docs/core/extensions/file-globbing.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/file-globbing.md) | [docs/core/extensions/file-globbing](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing?branch=pr-en-us-34538) |
| [docs/core/extensions/generic-host.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/generic-host.md) | [docs/core/extensions/generic-host](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/generic-host?branch=pr-en-us-34538) |
| [docs/core/extensions/globalization-and-localization.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/globalization-and-localization.md) | [docs/core/extensions/globalization-and-localization](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/globalization-and-localization?branch=pr-en-us-34538) |
| [docs/core/extensions/globalization-icu.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/globalization-icu.md) | [docs/core/extensions/globalization-icu](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu?branch=pr-en-us-34538) |
| [docs/core/extensions/globalization.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/globalization.md) | [docs/core/extensions/globalization](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/globalization?branch=pr-en-us-34538) |
| [docs/core/extensions/high-performance-logging.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/high-performance-logging.md) | [docs/core/extensions/high-performance-logging](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging?branch=pr-en-us-34538) |
| [docs/core/extensions/http-ratelimiter.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/http-ratelimiter.md) | [docs/core/extensions/http-ratelimiter](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/http-ratelimiter?branch=pr-en-us-34538) |
| [docs/core/extensions/httpclient-factory.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/httpclient-factory.md) | [docs/core/extensions/httpclient-factory](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory?branch=pr-en-us-34538) |
| [docs/core/extensions/httpclient-http3.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/httpclient-http3.md) | [docs/core/extensions/httpclient-http3](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-http3?branch=pr-en-us-34538) |
| [docs/core/extensions/localizability-review.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/localizability-review.md) | [docs/core/extensions/localizability-review](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/localizability-review?branch=pr-en-us-34538) |
| [docs/core/extensions/logging.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/logging.md) | [docs/core/extensions/logging](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/logging?branch=pr-en-us-34538) |
| [docs/core/extensions/options-library-authors.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/options-library-authors.md) | [docs/core/extensions/options-library-authors](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors?branch=pr-en-us-34538) |
| [docs/core/extensions/options.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/options.md) | [docs/core/extensions/options](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/options?branch=pr-en-us-34538) |
| [docs/core/extensions/timer-service.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/timer-service.md) | [docs/core/extensions/timer-service](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/timer-service?branch=pr-en-us-34538) |
| [docs/core/extensions/windows-service.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/windows-service.md) | [docs/core/extensions/windows-service](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/windows-service?branch=pr-en-us-34538) |
| [docs/core/extensions/workers.md](https://github.com/dotnet/docs/blob/013fb32189934393019ed98b19c2dfe0ecb1c461/docs/core/extensions/workers.md) | [docs/core/extensions/workers](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/workers?branch=pr-en-us-34538) |

</details>

This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/34538/checks).

### CODE REVIEW

File: docs/core/extensions/custom-logging-provider.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
 title: Implement a custom logging provider
-description: Learn how to implement a custom logging provider in your .NET applications.
+description: Discover how to implement a custom logging provider with colorized logs and custom logger in your .NET apps.
```
Comment (IEvangelist) [inline docs/core/extensions/custom-logging-provider.md:3]:
```suggestion
description: Discover how to implement a custom logging provider with colorized logs, writing custom ILogger and ILoggerProvider implementations.
```

File: docs/core/extensions/dependency-injection-usage.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ---
 title: Use dependency injection
-description: Learn how to use dependency injection in your .NET applications.
+description: Learn how to use dependency injection in your .NET apps with this comprehensive tutorial.
```
Comment (IEvangelist) [inline docs/core/extensions/dependency-injection-usage.md:3]:
```suggestion
description: Learn how to use dependency injection in your .NET apps with this comprehensive tutorial. Follow along with this pragmatic guide to understand DI in C#.
```

File: docs/core/extensions/dependency-injection.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ---
 title: Dependency injection
-description: Learn how .NET implements dependency injection and how to use it.
+description: Learn how to use dependency injection within your .NET apps. Discover how to registration services, define service lifetimes, and express dependencies.
```
Comment (IEvangelist) [inline docs/core/extensions/dependency-injection.md:3]:
```suggestion
description: Learn how to use dependency injection within your .NET apps. Discover how to registration services, define service lifetimes, and express dependencies in C#.
```

File: docs/core/extensions/timer-service.md [hunk 4]
```diff
@@ -3,7 +3,7 @@ title: Implement the IHostedService interface
 description: Learn how to implement a custom IHostedService interface with .NET.
```
Comment (IEvangelist) [inline docs/core/extensions/timer-service.md:3]:
```suggestion
description: Learn how to implement a custom IHostedService interface in C#, much like the inbuilt BackgroundService.
```

File: docs/core/extensions/workers.md [hunk 5]
```diff
@@ -3,7 +3,7 @@ title: Worker Services
 description: Learn how to implement a custom IHostedService and use existing implementations with .NET.
```
Comment (IEvangelist) [inline docs/core/extensions/workers.md:3]:
```suggestion
description: Learn how to implement a custom IHostedService and use existing implementations in C#. Discover various worker implementations, templates, and service patterns.
```

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

Let's :shipit:

**Pull request merged**